addons/account_edi/models/account_move.py

```diff
@@ -135,6 +136,169 @@ def _compute_edi_show_abandon_cancel_button(self):
+                quantity=invoice_line.quantity,
+                product=invoice_line.product_id,
+                partner=invoice_line.partner_id,
+                is_refund=invoice_line.move_id.move_type in ('in_refund', 'out_refund'),
+            )
+            rate = abs(invoice_line.balance) / abs(invoice_line.amount_currency) if invoice_line.amount_currency else 0.0
+            for tax_res in taxes_res['taxes']:
+                invoice_lines_tax_values_list.append({
+                    'base_line_id': invoice_line,
+                    'tax_id': self.env['account.tax'].browse(tax_res['id']),
+                    'tax_repartition_line_id': self.env['account.tax.repartition.line'].browse(tax_res['tax_repartition_line_id']),
+                    'base_amount': sign * invoice_line.company_currency_id.round(tax_res['base'] / rate if rate else 0.0),
+                    'tax_amount': sign * invoice_line.company_currency_id.round(tax_res['amount'] / rate if rate else 0.0),
+                    'base_amount_currency': sign * tax_res['base'],
+                    'tax_amount_currency': sign * tax_res['amount'],
+                })
+        grouping_key = grouping_key or default_grouping_key
+
+        # Apply 'applied_filter'.
+
+        if applied_filter:
+            invoice_lines_tax_values_list = [x for x in invoice_lines_tax_values_list if applied_filter(x)]
+
+        # Prepare the results.
+
+        res = {
+            'base_amount': 0.0,
+            'tax_amount': 0.0,
+            'base_amount_currency': 0.0,
+            'tax_amount_currency': 0.0,
+            'tax_details': defaultdict(lambda: {
+                'base_amount': 0.0,
+                'tax_amount': 0.0,
+                'base_amount_currency': 0.0,
+                'tax_amount_currency': 0.0,
+                'records': [],
+            }),
+            'invoice_line_tax_details': {
+                invoice_line: {
+                    'base_amount': invoice_line.balance,
+                    'tax_amount': 0.0,
+                    'base_amount_currency': invoice_line.amount_currency,
+                    'tax_amount_currency': 0.0,
+                    'tax_details': defaultdict(lambda: {
+                        'base_amount': 0.0,
+                        'tax_amount': 0.0,
+                        'base_amount_currency': 0.0,
+                        'tax_amount_currency': 0.0,
+                        'records': [],
+                    }),
+                } for invoice_line in invoice_lines
+            },
+        }
+
+        # Apply the 'grouping_key' method to 'invoice_lines_tax_values_list' and add all values to the final results.
+
+        seen_invoice_lines = set()
+        seen_grouped_base_lines = defaultdict(set)
+        for tax_values in invoice_lines_tax_values_list:
+            key = grouping_key(tax_values)
+            serialized_key = _serialize_python_dictionary(key)
+
+            invoice_line = tax_values['base_line_id']
```

Comment:
WIthin such a long fuction, I'd avoid reusing variable names, to avoid confusion. base_line ?